src/main/java/slimeknights/tconstruct/library/client/RenderUtil.java

```diff
@@ -1,12 +1,342 @@
 package slimeknights.tconstruct.library.client;
 
+import com.mojang.blaze3d.matrix.MatrixStack;
 import com.mojang.blaze3d.systems.RenderSystem;
+import com.mojang.blaze3d.vertex.IVertexBuilder;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.Matrix4f;
+import net.minecraft.client.renderer.RenderHelper;
+import net.minecraft.client.renderer.RenderState;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.client.renderer.texture.AtlasTexture;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.util.Direction;
+import net.minecraftforge.fluids.FluidStack;
+import org.lwjgl.opengl.GL11;
+import slimeknights.tconstruct.TConstruct;
 
 public final class RenderUtil {
 
   private RenderUtil() {
   }
 
+  public static final float FLUID_OFFSET = 0.005f;
+  public static final RenderType FLUID_RENDER_TYPE = RenderType.makeType(TConstruct.modID + ":fluid_render_type",
+    DefaultVertexFormats.POSITION_COLOR_TEX_LIGHTMAP, 7, 256, true, false,
+    RenderType.State.getBuilder().texture(new RenderState.TextureState(AtlasTexture.LOCATION_BLOCKS_TEXTURE, false, false))
+      .shadeModel(RenderType.SHADE_ENABLED)
+      .lightmap(RenderType.LIGHTMAP_ENABLED)
+      .texture(RenderType.BLOCK_SHEET_MIPPED)
+      .transparency(RenderType.TRANSLUCENT_TRANSPARENCY)
+      .build(false));
+
+  protected static Minecraft mc = Minecraft.getInstance();
+
+  /**
+   * Renders a fluid block, call from TESR. x/y/z is the rendering offset.
+   *
+   * @param fluid Fluid to render
+   * @param w     Width. 1 = full X-Width
+   * @param h     Height. 1 = full Y-Height
+   * @param d     Depth. 1 = full Z-Depth
+   */
+  public static void renderFluidCuboid(FluidStack fluid, MatrixStack matrices, IVertexBuilder renderer, int combinedLight, float w, float h, float d) {
+    float wd = (1f - w) / 2f;
+    float hd = (1f - h) / 2f;
+    float dd = (1f - d) / 2f;
+
+    renderFluidCuboid(fluid, matrices, renderer, combinedLight, wd, hd, dd, 1f - wd, 1f - hd, 1f - dd);
+  }
+
+  public static void
+  renderFluidCuboid(FluidStack fluid, MatrixStack matrices, IVertexBuilder renderer, int combinedLight, float x1, float y1, float z1, float x2, float y2, float z2) {
```

Comment:
Formatting got weird here. And javadocs (you can borrow from the one above, just explain the differences at the start, such as defaults)

Reply:
@shedaniel ^